src/bandersnatch_filter_plugins/metadata_filter.py

```diff
@@ -176,14 +179,83 @@ def filter(self, metadata: Dict) -> bool:
         return RegexFilter.filter(self, metadata)
 
 
+class SizeProjectMetadataFilter(FilterMetadataPlugin, AllowListProject):
+    """
+    Plugin to download only packages having total file sizes less than
+    a configurable threshold.
+    """
+
+    name = "size_project_metadata"
+    initialized = False
+    max_package_size: int = 0
+    allowlist_package_names: List[str] = []
+
+    def initialize_plugin(self) -> None:
+        """
+        Initialize the plugin reading settings from the config.
+        """
+        if not self.initialized:
+            try:
+                human_package_size = self.configuration["size_project_metadata"][
+                    "max_package_size"
+                ]
+            except KeyError:
+                logger.warn(
+                    f"Unable to initialise {self.name} plugin;"
+                    f"must create max_package_size in configuration."
+                )
+                return
+            try:
+                self.max_package_size = parse_size(human_package_size, binary=True)
+            except InvalidSize:
+                logger.warn(
+                    f"Unable to initialise {self.name} plugin;"
+                    f'max_package_size of "{human_package_size}" is not valid.'
+                )
+                return
+            if self.max_package_size > 0:
+                if not self.allowlist_package_names:
+                    self.allowlist_package_names = (
+                        self._determine_unfiltered_package_names()
+                    )
+
+                logger.info(
+                    f"Initialized metadata plugin {self.name} to block projects >{self.max_package_size}b"  # noqa: E501
```

Comment:
I guess the E501 is to keep your local editor happy?

Reply:
Just a line too long. I'll split it if desired, but I thought it might get a bit confusing with the ternary if-else following.